investor_wallet_platform_base/models/subscription_request.py

```diff
@@ -38,31 +38,32 @@ def get_mail_template_notif(self, is_company=False):
                 "sub_req_notif", self.structure
             )
 
-    def get_capital_release_mail_template(self):
-        template_obj = self.env["mail.template"]
-        return template_obj.get_email_template_by_key(
-            "rel_capital", self.structure
-        )
+    def _send_confirmation_email(self):
+        if not self.structure.is_delegated_to_api_client:
+            return super()._send_confirmation_email()
+
+    def _send_waiting_list_email(self):
+        if not self.structure.is_delegated_to_api_client:
+            return super()._send_waiting_list_email()
+
+    def send_new_subscription_request_notification_email(self, is_company):
+        if not self.structure.is_delegated_to_api_client:
+            notification_template = self.get_structure_email_template_notif(
+                is_company=self.partner_id.is_company
+            )
+            notification_template.send_mail(self.id)
 
     @api.model
     def create(self, vals):
-        subscr_request = super(SubscriptionRequest, self).create(vals)
-        notification_template = subscr_request.get_structure_email_template_notif(
-            False
-        )
-        notification_template.send_mail(subscr_request.id)
-        return subscr_request
+        subscription_request = super(SubscriptionRequest, self).create(vals)
+        subscription_request.send_new_subscription_request_notification_email(is_company=False)
+        return subscription_request
 
     @api.model
     def create_comp_sub_req(self, vals):
-        subscr_request = super(SubscriptionRequest, self).create_comp_sub_req(
-            vals
-        )
-        notification_template = subscr_request.get_structure_email_template_notif(
-            True
-        )
-        notification_template.send_mail(subscr_request.id)
-        return subscr_request
+        subscription_request = super(SubscriptionRequest, self).create_comp_sub_req(            vals        )
```

Comment:
not pep 8

Reply:
![](https://media.giphy.com/media/RFDXes97gboYg/giphy.gif)